File: LeetCode/398.Random_Pick_Index/398.Random_Pick_Index.py
#!/usr/bin/env python3


class Solution:

    def __init__(self, nums: List[int]):
        # dic key:num, val:index array
        self.dic = defaultdict(list)
        for i, num in enumerate(nums):
            self.dic[num].append(i)

    def pick(self, target: int) -> int:
        if target in self.dic:
            return random.choice(self.dic[target])
        return -1


# Your Solution object will be instantiated and called as such:
# obj = Solution(nums)
# param_1 = obj.pick(target)

File: LeetCode/398.Random_Pick_Index/398.Random_Pick_Index.swift

class Solution {
    var dic: [Int: [Int]] = [:]

    init(_ nums: [Int]) {
        for (i, num) in nums.enumerated() {
            dic[num, default: []].append(i)
        }
    }

    func pick(_ target: Int) -> Int {
        if let arr = dic[target] {
            let i = Int.random(in: 0 ..< arr.count)
            return arr[i]
        }
        return -1
    }
}

/**
 * Your Solution object will be instantiated and called as such:
 * let obj = Solution(nums)
 * let ret_1: Int = obj.pick(target)
 */
